### DISCUSSION TITLE
Running x86 images on ARM nodes

### DISCUSSION THREAD

**Discussion by GlebBeloded** [k3s-io/k3s#10276] [Q&A]
Hi! We have been using "docker build && docker push" to build x86 images.
Somehow, k3s cluster was able to run those images on k3s-arm64 on m1 macbooks.

After switching to buildkit, k3s (with containerd inside) is no longer able to run these images with an error:
`"foo.bar/img":  no match for platform in manifest: not found`

I've been trying to google it and find out when and why compatibility broke, but was no successful in my search.
Old images had ` "DockerVersion": "24.0.5"`, new ones (which are not  working) have ` "Comment": "buildkit.dockerfile.v0"`.

Interestingly, when I run image via ctr from k3s node, it runs without problem.
Is there a way to bypass this limitation somehow, or I am forced to build for two architectures on newer buildkit/docker versions?
Thank you!

**Comment by brandond** [comment]
Previously you were building and tagging individual images, and the platform in the image's config was ignored. Now with buildkit your tags point at multi-arch manifest lists, and the attempt to run the image is rejected earlier, as it can't find an image that matches your platform in the manifest list.

**Comment by hinst** [comment]
Greetings from 2026, the compatibility problem described in the post still exists. Might someone have a solution? As pointed out, x86 emulation was working fine before, therefore it is possible with right configuration
